app/routes.py

```diff
@@ -1,2 +1,314 @@
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+
+        db.session.commit()
+
+        committed_goal = {"goal": 
+            {"id": goal.goal_id,
+            "title": goal.title,
+            }}
+        return jsonify(committed_goal), 200
+    
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        goal_response_body = {
+            "details":
+                f'Goal {goal.goal_id} successfully deleted'
+            }
+        return jsonify(goal_response_body)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])
+def handle_goal_tasks(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+
+    if request.method == "GET":
+        tasks = goal.tasks
+        list_of_tasks = []
+        
+        for task in tasks:
+            if task.completed_at == None:
+                completed_at = False
+            else:
+                completed_at = True
+
+            individual_task = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": completed_at,
+                "goal_id": goal.goal_id
+            }
+            list_of_tasks.append(individual_task)
+
+        return make_response({
+            "id": goal.goal_id,
+            "title": goal.title,
+            "tasks": list_of_tasks
+        })
+
+    if request.method == "POST":
+        goal = Goal.query.get(goal_id)
+        request_body = request.get_json()
+
+        for ids_per_task in request_body["task_ids"]:
+            task = Task.query.get(ids_per_task)
+            goal.tasks.append(task)
+            db.session.add(goal)
+            db.session.commit()
+
+        return make_response({
+            "id": goal.goal_id,
+            "task_ids": request_body["task_ids"]
+        })
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "POST": 
+        request_body = request.get_json()
+        title = request_body.get("title")
+        description = request_body.get("description")
+
+        if not title or not description or "completed_at" not in request_body:
+            return make_response(jsonify({"details": "Invalid data"}), 400)
+
+        new_task = Task(
+                    title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])
+
+        if new_task.completed_at == None:
+            completed_at = False
+        else:
+            completed_at = True
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return make_response({
+            "task": {
+                "id": new_task.task_id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": completed_at
+            }}, 201)
```

Comment:
Just like we talked about in the review session, how could you create a helper function instead of having to write this every time?